src/legacy/ui/public/chrome/directives/header_global_nav/components/header.tsx

```diff
@@ -235,31 +238,28 @@ class HeaderUI extends Component<Props, State> {
     const leftNavControls = navControls.bySide[NavControlSide.Left];
     const rightNavControls = navControls.bySide[NavControlSide.Right];
 
-    let navLinksArray = navLinks.map(navLink =>
-      navLink.hidden || !uiCapabilities.navLinks[navLink.id]
-        ? null
-        : {
-            key: navLink.id,
-            label: navLink.title,
-            href: navLink.href,
-            iconType: navLink.euiIconType,
-            icon:
-              !navLink.euiIconType && navLink.icon ? (
-                <EuiImage
-                  size="s"
-                  alt=""
-                  aria-hidden={true}
-                  url={chrome.addBasePath(`/${navLink.icon}`)}
-                />
-              ) : (
-                undefined
-              ),
-            isActive: navLink.active,
-            'data-test-subj': 'navDrawerAppsMenuLink',
-          }
-    );
-    // filter out the null items
-    navLinksArray = navLinksArray.filter(item => item !== null);
+    const navLinksArray = navLinks
+      .filter(navLink => !navLink.hidden)
+      .map(navLink => ({
+        key: navLink.id,
+        label: navLink.title,
+        href: addBasePath(navLink.href),
```

Comment:
This is necessary now because extendNavLink is no longer setting the actual href value but including the `appUrl` sometimes (which doesn't have the basePath) and absolute urls the rest of the time. I feel like we probably don't need both of these.